DragaliaAPI.Test/Unit/Controllers/DungeonRecordControllerTest.cs

```diff
@@ -0,0 +1,579 @@
+﻿using DragaliaAPI.Controllers.Dragalia;
+using DragaliaAPI.Database.Entities;
+using DragaliaAPI.Database.Repositories;
+using DragaliaAPI.Models;
+using DragaliaAPI.Models.Generated;
+using DragaliaAPI.Services;
+using DragaliaAPI.Shared.Definitions.Enums;
+using DragaliaAPI.Shared.MasterAsset.Models;
+using FluentAssertions.Execution;
+using Microsoft.AspNetCore.Mvc;
+using MockQueryable.Moq;
+using static DragaliaAPI.Test.TestUtils;
+
+namespace DragaliaAPI.Test.Unit.Controllers;
+
+public class DungeonRecordControllerTest
+{
+    private readonly DungeonRecordController dungeonRecordController;
+    private readonly Mock<IQuestRepository> mockQuestRepository;
+    private readonly Mock<IDungeonService> mockDungeonService;
+    private readonly Mock<IUserDataRepository> mockUserDataRepository;
+    private readonly Mock<IInventoryRepository> mockInventoryRepository;
+    private readonly Mock<IUpdateDataService> mockUpdateDataService;
+
+    private const string dungeonKey = "key";
+    private const int questId = 320180103;
+    private const int clearTime = 60;
+
+    public DungeonRecordControllerTest()
+    {
+        this.mockQuestRepository = new(MockBehavior.Strict);
+        this.mockDungeonService = new(MockBehavior.Strict);
+        this.mockUserDataRepository = new(MockBehavior.Strict);
+        this.mockInventoryRepository = new(MockBehavior.Strict);
+        this.mockUpdateDataService = new(MockBehavior.Strict);
+
+        this.dungeonRecordController = new(
+            mockQuestRepository.Object,
+            mockDungeonService.Object,
+            mockUserDataRepository.Object,
+            mockInventoryRepository.Object,
+            mockUpdateDataService.Object
+        );
+
+        this.dungeonRecordController.SetupMockContext();
+
+        this.mockDungeonService
+            .Setup(x => x.FinishDungeon(dungeonKey))
+            .ReturnsAsync(
+                new DungeonSession()
+                {
+                    Party = new List<PartySettingList>(),
+                    QuestData = new QuestData(
+                        Id: questId,
```

Comment:
Could you also add a test to make sure that the party and quest id values appear in the response as expected? Search for quest_party_setting_list and quest_id